### PR TITLE
geolocation: update to use GeoJSON

### LINKED ISSUE #73: `lat` and `long` in Annotations
Right now, `latitude` and `longitude` are in the `core` namespace for Annotations segments. This was put in *before* we had figured out the continuously variable method for annotating geolocation (per #55).

I think this opens the question, do we still want `lat` and `long` in the `core` namespace as a field in one of the segments? And, if so, does it belong in Annotations?

We *know* that many people will be using the new multiple-recordings approach to record geolocation for a recording. The downside, if that is the only way to define geolocation, is that we create a lot of overhead if you have a stationary receiver that you want to provide the coordinates for. The upside, though, is that there is no risk of confusion regarding what to do if both a Recording of geolocation and `lat` & `long` tags exist.

Here is my suggestion: In the `global` namespace, we add a `core:geolocation` field. This field can either be a formatted lat/long coordinate, in which case it's static for the recording, or it uses whatever method for linking to another recording that we land on in #55, in which case it's obvious that it's volatile.

Thoughts?

### LINKED ISSUE #84: 'altitude' field, and being able to put coordinates in global
I reviewed the current version of SigMF to see if it would work for what I'm doing, and the only thing that would be nice to add is an optional altitude field after the latitude and longitude.  I mean it's optional so people without altitude can just leave it empty.

Another aspect that seems a little weird is that I can't just put the coordinates in global, which applies to the entire dataset.  For what I'm doing there's a set of coordinates that goes along with every file.  With the current SigMF I suppose I can just add an annotation to the first sample in each file, with a sample_count of either 1 or the number of samples in the dataset.  I guess that works, it just feels a little weird.  I think a lot of applications involve saving coordinates that don't change over a file and may never change =)

Also I think it was already mentioned but we need to decide on a standard coordinate system/units.  Doesn't matter, just needs to be standardized to save headache.

### PR DESCRIPTION
**Pull request by jacobagilbert**
This closes #73 and closes #84 by identifying how altitude data should be included in recordings, and codifies the use of GeoJSON for the new `global` object `core:geolocation`.

This also marks the `core:latitude` and `core:longitude` fields in `annotations` for deprecation in a future (breaking) release.

### CODE REVIEW

**Review by Teque5 — CHANGES_REQUESTED** [review]
This looks good, but I think we should remove the `latitude` and `longitude` keys instead of marking them as depricated.

Also remember to update the `schema.json`. I think also in `validate.py:validate_key()` you will have to add the `geometry` type.

**Comment by jacobagilbert** [comment]
@Teque5 In general I agree but removing latitude and longitude keys would be a breaking change, and if I am recalling correctly (@bhilburn help me out here) we have decided that v1.0 should not break the v0.x spec (and I am aware of current software products using them). I need to update this per discussion in #73 and I'll update the other things you pointed out, thanks.

File: sigmf-spec.md
Comment (bhilburn) [inline sigmf-spec.md:308]: I think we should add detail that makes it clear that:
* those fields must appear in that order
* they are all WGS-84
* this is defined by [RFC 5870](https://datatracker.ietf.org/doc/html/rfc5870), which GeoJSON implements

Also, you've got a typo one line up for `latitude`

File: sigmf-spec.md
Comment (bhilburn) [inline sigmf-spec.md:298]: Is "HAGL" described meaningfully somewhere that we could refer to? Honest question. @dharasty?

File: sigmf-spec.md
Comment (bhilburn) [inline sigmf-spec.md:297]: I think we need to be SUPER clear that the only thing supported is a `GeoJSON **Point**`

**Review by bhilburn — CHANGES_REQUESTED** [review]
Great PR, @jacobagilbert. I left a lot of comments / questions, but that's because this is a super complicated topic and I want to make sure we cover the edge cases everywhere we can. It is basically closing 4 years of discussion, so, not unexpected 😄

@Teque5 - I agree, and would prefer to just remove those deprecated fields entirely. The limiting thing, here, is that we do want to maintain backwards compatibility between v1.0.0 and v0.0.1, because we know v.0.0.1 is so widely used, and we don't want to break all those systems without giving them a stably versioned release.

File: sigmf-spec.md
Comment (jacobagilbert) [reply sigmf-spec.md:298]: "...(in meters) as measured by Ottoman calipers from the phase center of the aperture to the mean surface level of the normal square decimeter in the direction of the center of the WGS84 ellipse"

:)

File: sigmf-spec.md
Comment (jacobagilbert) [reply sigmf-spec.md:297]: Agree. I will fix this.

File: sigmf-spec.md
Comment (jacobagilbert) [reply sigmf-spec.md:304]: Yes, this makes sense and I agree. #99 can be rebased off this (needs changes anyway).

File: sigmf-spec.md
Comment (jacobagilbert) [reply sigmf-spec.md:305]: agree.

File: sigmf-spec.md
Comment (jacobagilbert) [reply sigmf-spec.md:308]: Agree (and thanks. damn gedit's lack of spell checking)

**Comment by jacobagilbert** [comment]
@bhilburn I *think* I have resolved what you were asking about (ok..not HAGL i'll let someone else chime in there...). I reference the RFC texts in only one place and used the "`point` object" language anywhere it is described to make sure that is clear. If something here isn't resolved let me know.

**Comment by gmabey** [comment]
@777arc are you able to review this?

File: sigmf-spec.md
Comment (777arc) [reply sigmf-spec.md:298]: Did we want that extended description of HAGL added to line 298?  I don't feel strongly about it either way.

**Review by 777arc — APPROVED** [review]
LGTM, I added a note about HAGL but don't let that keep it from getting merged in

File: sigmf-spec.md
Comment (dharasty) [reply sigmf-spec.md:298]: > Is "HAGL" described meaningfully somewhere that we could refer to? Honest question. @dharasty?

Alas, I search high and low for something like an authoritative glossary (such as an FCC document) and -- to my amazement -- couldn't find one.

Comment (jacobagilbert) [reply sigmf-spec.md:298]: I believe that HAGL is a generally informal term, I find what is in the PR to be sufficient.

**Comment by bhilburn** [comment]
@jacobagilbert - New changes look great, but merge is blocked by conflict. Can you push an update resolving the conflict? Also, since you'll be in there, can you make your reference to RFC 7946 actually link to that IETF doc?

Once merge conflict is resolved, we are good to go here 😁

**Comment by jacobagilbert** [comment]
Should be G2G @bhilburn if you want to take one more quick look at the schema since I had to manually resolve that conflict that would be great.

**Pull request merged**
